File: 2022/02/part1.rb
# frozen_string_literal: true

data = File.read(
  File.join(
    File.dirname(__FILE__),
    './input.txt'
  )
).split("\n")

# R, P, S

def score(a, b)
  if (b == 'X' && a == 'C') || (b == 'Y' && a == 'A') || (b == 'Z' && a == 'B')
    6
  elsif (b == 'X' && a == 'A') || (b == 'Y' && a == 'B') || (b == 'Z' && a == 'C')
    3
  else
    0
  end
end

def type(b)
  case b
  when 'X'
    1
  when 'Y'
    2
  else
    3
  end
end

sum = 0

data.each do |line|
  a,b = line.split
  sum += score(a,b) + type(b)
end

print sum

File: 2023/01/part1.exs
file = 'input.txt'
input = File.read!(Path.join(Path.dirname(__ENV__.file), file))
lines = String.split(input)

first_and_last_digit_combined = fn line ->
  matches = List.flatten(Regex.scan(~r/[0-9]/, line))

  fe = hd(matches)
  le = hd(Enum.reverse(matches))
  concat = fe <> le
  String.to_integer(concat)
end

numbers = Enum.map(lines, first_and_last_digit_combined)

IO.puts(numbers |> Enum.sum)
